testing/e2e/e2e_api_test.go

```diff
@@ -21,14 +21,76 @@
 package e2e_test
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
 	beaconapi "github.com/attestantio/go-eth2-client/api"
 	apiv1 "github.com/attestantio/go-eth2-client/api/v1"
 	"github.com/attestantio/go-eth2-client/spec/phase0"
 	"github.com/berachain/beacon-kit/node-api/handlers/utils"
 	"github.com/berachain/beacon-kit/testing/e2e/config"
 	"github.com/berachain/beacon-kit/testing/e2e/suite/types"
+	"github.com/ethereum/go-ethereum/common"
 )
 
+// StateValidatorResponse represents the API response for a validator.
+type StateValidatorResponse struct {
+	Data struct {
+		Index     string `json:"index"`
+		Balance   string `json:"balance"`
+		Status    string `json:"status"`
+		Validator struct {
+			Pubkey string `json:"pubkey"`
+		} `json:"validator"`
+	} `json:"data"`
+}
+
+// BeaconHTTPClient wraps http.Client with baseURL.
+// This is needed to change the default baseURL of the http.Client.
+type BeaconHTTPClient struct {
+	*http.Client
+	baseURL string
+}
+
+// Get overrides http.Client.Get to use baseURL
+func (c *BeaconHTTPClient) Get(path string) (*http.Response, error) {
+	return c.Client.Get(c.baseURL + path)
+}
+
+// pubkey is the public key of the validator.
+// Made it global so that I can use it in another test.
+var pubkey string
```

Comment:
I don't understand why this needs to be reused in test. Would prefer not having global vars.

Reply:
Also nit: for code comments, let's try not to use "I"; "we" is sometimes okay but code comments generally shouldn't be written in the [first-person](https://www.grammarly.com/blog/grammar/first-second-and-third-person/)

Reply:
The idea was to query the validator by the public key. Removed the global vars, re-written the test in another way.

Thanks, never do that. This auto comment :D